### ISSUE TITLE
[2.0.x] PDO::FETCH_*  (Need clarification)

### ISSUE THREAD

**Issue by sergeyklay** [phalcon/cphalcon#10362]
What is the difference between:
```zephir
public function $fetch()
{
    return this->_pdoStatement->$fetch();
}

and

public function fetchArray()
{
    return this->_pdoStatement->$fetch();
}
```

Also current [setFetchMode](https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/db/result/pdo.zep#L307) method is not quite correct:

See http://php.net/manual/en/pdostatement.setfetchmode.php

**Comment by andresgutierrez** [comment]
It seems setFetchMode needs to match the PDO one

**Comment by andresgutierrez** [comment]
This is fixed in 2.0.4 (2.0.x branch)
